cli/src/commonMain/kotlin/StartCommand.kt

```diff
@@ -100,15 +113,98 @@ class StartCommand : SuspendingCliktCommand(name = "start") {
         ContextStorage.saveLatestRunId(ortRun.id)
 
         if (wait) {
-            while (ortRun.isRunning()) {
-                delay(POLL_INTERVAL)
-                ortRun = client.repositories.getOrtRun(repositoryId, ortRun.index)
-                if (!useJsonFormat) echoMessage(ortRun.toPrintable())
+            if (useJsonFormat) {
+                while (ortRun.isRunning()) {
+                    delay(POLL_INTERVAL)
+                    ortRun = client.repositories.getOrtRun(repositoryId, ortRun.index)
+                }
+            } else {
+                echoMessage(ortRun.toPrintable())
+                ortRun = updateProgressBar(ortRun, client)
             }
         }
 
         echoMessage(ortRun.toPrintable())
     }
+
+    private fun updateProgressBar(ortRun: OrtRun, client: OrtServerClient): OrtRun = runBlocking {
+        val maxProgressBarValue = JobType.entries.count { it.isConfigured(ortRun.jobConfigs) }.withInitialState()
+
+        val progress = progressBarContextLayout {
+            text { context }
+            spinner(Spinner.Dots())
+            progressBar(width = 30)
+            percentage()
+        }.animateInCoroutine(
+            terminal,
+            context = "Starting",
+            total = maxProgressBarValue,
+            completed = 0
+        )
+
+        launch { progress.execute() }
+
+        var updatedOrtRun = ortRun.copy()
+        while (updatedOrtRun.isRunning()) {
+            delay(POLL_INTERVAL)
+            updatedOrtRun = client.repositories.getOrtRun(repositoryId, updatedOrtRun.index)
+            progress.update {
+                context =
+                    JobType.entries.reversed().find { it.isStarted(updatedOrtRun.jobs) }?.displayName ?: "Starting"
+                completed = JobType.entries.count { it.isFinished(updatedOrtRun.jobs) }.withInitialState()
+            }
+        }
+
+        progress.update {
+            context = "Finished"
+            completed = maxProgressBarValue
+        }
+
+        updatedOrtRun
+    }
+
+    private enum class JobType(val displayName: String) {
+        ANALYZER("Analyzing"),
+        ADVISOR("Advising"),
+        SCANNER("Scanning"),
+        EVALUATOR("Evaluating"),
+        REPORTER("Reporting"),
+        NOTIFIER("Notifying");
+
+        fun isConfigured(config: JobConfigurations) = when (this) {
+            ANALYZER -> true
+            ADVISOR -> config.advisor != null
+            SCANNER -> config.scanner != null
+            EVALUATOR -> config.evaluator != null
+            REPORTER -> config.reporter != null
+            NOTIFIER -> config.notifier != null
+        }
+
+        fun isStarted(jobs: Jobs) = when (this) {
+            ANALYZER -> true
```

Comment:
When a run has no Analyzer Job (for instance when the user provides an ort result), is this logic still going to work ?

Reply:
At the current state, it is not possible to have an ORT Server run without an analyzer.

> for instance when the user provides an ort result

Are you sure that this feature exists in the ORT Server? Can you point me to it?

Reply:
No it does not exist but I believe it was on our roadmap.

Reply:
See https://github.com/eclipse-apoapsis/ort-server/issues/193.

Reply:
The assumption is correct for the `isConfigured` function, but not for the `isStarted` function.
FIxed.